sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcTestDataSet.java

```diff
@@ -0,0 +1,128 @@
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Date;
+import javax.sql.DataSource;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.postgresql.ds.PGSimpleDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manipulates test data used by the {@link org.apache.beam.sdk.io.jdbc.JdbcIO} tests.
+ *
+ * <p>This is independent from the tests so that for read tests it can be run separately after data
+ * store creation rather than every time (which can be more fragile.)
+ */
+public class JdbcTestDataSet {
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcTestDataSet.class);
+  public static final String[] SCIENTISTS = {"Einstein", "Darwin", "Copernicus", "Pasteur", "Curie",
+      "Faraday", "McClintock", "Herschel", "Hopper", "Lovelace"};
+  /**
+   * Use this to create the read tables before IT read tests.
+   *
+   * <p>To invoke this class, you can use this command line:
+   * mvn test-compile exec:java -Dexec.mainClass=org.apache.beam.sdk.io.jdbc.JdbcTestDataSet \
+   *   -Dexec.args="--postgresIp=1.1.1.1 --postgresUsername=postgres
+   *   --postgresDatabaseName=myfancydb \
+   *   --postgresPassword=yourpassword --postgresSsl=false" \
+   *   -Dexec.classpathScope=test
+   * @param args Please pass options from PostgresTestOptions used for connection to postgres as
+   * shown above.
+   */
+  public static void main(String[] args) throws SQLException {
+    PipelineOptionsFactory.register(PostgresTestOptions.class);
+    PostgresTestOptions options =
+        PipelineOptionsFactory.fromArgs(args).as(PostgresTestOptions.class);
+
+    createReadDataTable(getDataSource(options));
+  }
+
+  public static PGSimpleDataSource getDataSource(PostgresTestOptions options) throws SQLException {
+    PGSimpleDataSource dataSource = new PGSimpleDataSource();
+
+    // Tests must receive parameters for connections from PipelineOptions
+    // Parameters should be generic to all tests that use a particular datasource, not
+    // the particular test.
+    dataSource.setDatabaseName(options.getPostgresDatabaseName());
+    dataSource.setServerName(options.getPostgresServerName());
+    dataSource.setPortNumber(options.getPostgresPort());
+    dataSource.setUser(options.getPostgresUsername());
+    dataSource.setPassword(options.getPostgresPassword());
+    dataSource.setSsl(options.getPostgresSsl());
+
+    return dataSource;
+  }
+
+  public static final String READ_TABLE_NAME = "BEAM_TEST_READ";
+
+  public static void createReadDataTable(DataSource dataSource) throws SQLException {
+    createDataTable(dataSource, READ_TABLE_NAME);
+  }
+
+  public static String createWriteDataTable(DataSource dataSource) throws SQLException {
+    String tableName = "BEAMTEST".concat(new Long(new Date().getTime()).toString());
```

Comment:
`org.joda.time.Instant.now().getMillis()`? You are likely already depending on the JodaTime libraries, as does the rest of the codebase.

Also, I would strongly prefer either using `String.format` or `+`-based concatenation rather than `String#concat`, again for consistency